components/eam/src/chemistry/yaml/modal_aer_opt/f90_yaml/modal_aero_lw_beg_yml.f90

```diff
@@ -0,0 +1,133 @@
+
+  !-----------------------------------------------------------------------------------------
+  !In the case of y_i or y_k are not passed as arguments, use the following if condition:
+  if(yaml%col_print >0 .and. y_nstep==yaml%nstep_print) then
+  !-----------------------------------------------------------------------------------------
+
+  !-----------------------------------------------------------------------------------------
+  !For generating data for a dependent subroutines where "yaml" derived type is already
+  !initialized, use the following if condition
+  !if(yaml%flag_print) then
+  !-----------------------------------------------------------------------------------------
+
+     !-----------------------------------------------------------------------------------------
+     ! Set "ext_str" if there are multiple sets of yaml output to be written out
+     ! Example:"flag" in the code can be 0, 1, or 2, we can update "ext_str" as:
+     ! write(ext_str,'(I2)') flag
+     ! ext_str = 'flag_'//adjustl(ext_str)
+     !-----------------------------------------------------------------------------------------
+
+
+     !Record number of calls that can output yaml file if you only need to write one set of input/output
+     n_calls = n_calls+1
+
+     if (n_calls==1) then ! output at the first call only, modify this if condition (see below) if writing out other calls
+
+     !if ((n_calls==1 .and. flag==0) .or. (n_calls==3 .and. flag==1) .or. (n_calls==5 .and. flag==2)) then
+
+        !(**remove these yaml% variables if generating data for a dependent subroutines**)
+        yaml%lchnk_print = y_lchnk
+        yaml%flag_print  = .true.
+
+
+        !open I/O yaml files
+        !(with an optional argument to pass a unique string to differentiate file names)
+        call open_files('modal_aero_lw', &  !intent-in
+             unit_input, unit_output) !intent-out
+        !    unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print,yaml%col_print, &
+             'modal_aero_lw',yaml%nstep_print, yaml%lev_print)
+
+        ! add code for writing data here
+        
+        call write_var(unit_input,unit_output,'dt',dt)
+        call write_var(unit_input,unit_output,'ncol',ncol)
+        call write_var(unit_input,unit_output,'state_q',state_q(yaml%col_print,:,:))
+        call write_var(unit_input,unit_output,'temperature',temperature(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pmid',pmid(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pdel',pdel(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'pdeldry',pdeldry(yaml%col_print,:))
+        call write_var(unit_input,unit_output,'cldn',cldn(yaml%col_print,:))
```

Comment:
please add: `refrtablw`, `refitablw`, and `specrefndxlw`; [see here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/966023e8429c1f913df38743dc84a1ab9911406b/components/eam/src/physics/cam/modal_aer_opt.F90#L32)